### DISCUSSION TITLE
Error when running MRSimulator for the first time

### DISCUSSION THREAD

**Discussion by mjunior-NMR** [deepanshs/mrsimulator#294] [Q&A] [answered]
Dear mrsimulator comunity, 
I am new to python and specially new to MRSimulator. Please, forgive me if my question is too naive. After installing the package with pip install, I've tried the first example code on the [MRSimulator website](https://mrsimulator.readthedocs.io/en/stable/installation/users.html) but got the error below. Actually, only by runing import mrsimulator I already have the same error messages. I would appreciate it If someone could help me to get started.

`Traceback (most recent call last):

  File C:\ProgramData\Anaconda3\envs\mojrmn\lib\site-packages\spyder_kernels\py3compat.py:356 in compat_exec
    exec(code, globals, locals)

  File d:\marcos\ifsc\python\mrsimulator\my_first_sim.py:7
    import mrsimulator as mrs

  File ~\AppData\Roaming\Python\Python39\site-packages\mrsimulator\__init__.py:72
    from .simulator import Simulator  # lgtm [py/import-own-module] # noqa:F401

  File ~\AppData\Roaming\Python\Python39\site-packages\mrsimulator\simulator\__init__.py:15
    from mrsimulator.method import Method

  File ~\AppData\Roaming\Python\Python39\site-packages\mrsimulator\method\__init__.py:21
    from .event import MixingEvent  # noqa: F401

  File ~\AppData\Roaming\Python\Python39\site-packages\mrsimulator\method\event.py:172
    class ConstantDurationEvent(BaseEvent):  # TransitionModulationEvent

  File ~\AppData\Roaming\Python\Python39\site-packages\mrsimulator\method\event.py:216 in ConstantDurationEvent
    **BaseEvent().property_default_units,

  File ~\AppData\Roaming\Python\Python39\site-packages\pydantic\main.py:165 in __init__
    __pydantic_self__.__pydantic_validator__.validate_python(data, self_instance=__pydantic_self__)

ValidationError: 1 validation error for BaseEvent
rotor_frequency
  Input should be a valid number [type=float_type, input_value=None, input_type=NoneType]
    For further information visit https://errors.pydantic.dev/2.3/v/float_type`

Thanks a lot!

**Comment by mjunior-NMR** [comment]
Solved: 

In \python\python39\site-packages\mrsimulator\method I have changed the file "event.py" on line 216:

```
property_units: Dict = {
        "duration": "µs",
        ****BaseEvent().property_default_units, #I've changed to **BaseEvent.property_default_units, (removed the parenthesis).**
    }
```

**Reply by deepanshs** [reply]
You shouldn't have to update the code in the site-packages. It is likely an error from a newer version of pydantic. Can you try installing pydantic v1.10? The next release of mrsimulator will have a fix for version incompatibility.

**Comment by deepanshs** [comment]
Hi,

I have noticed this error before. It is likely from the newer version of pydantic. Can you check the version of pydantic installed on your system. `pip show pydantic`

**Reply by mjunior-NMR** [reply, marked as answer]
Dear Deepansh, thank you very much for your help. Installing pydantic version 1.10 solved the issue. 
Best regards, 
Marcos.
